### ISSUE TITLE
panic: runtime error: invalid memory address or nil pointer dereference

### ISSUE THREAD

**Issue by cnslps** [OWASP/Amass#461]
```
$ amass -version
v3.8.2
$ 
```

`$ amass enum -d xxx.xxx.bd -ip -o amassout.txt -max-dns-queries 1000 -brute -w bdwords.txt`

```
The enumeration has finished
Discoveries are being migrated into the Cayley Graph database
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x50 pc=0xbd5e46]

goroutine 74 [running]:
github.com/yuin/gopher-lua.(*LState).PCall.func1(0x1299c08, 0xc0086ec0b0, 0xc00351fab8, 0x0, 0x0, 0x0)
	/Users/caffix/go/pkg/mod/github.com/yuin/gopher-lua@v0.0.0-20200603152657-dc2b0ca8b37e/state.go:2007 +0x116
panic(0xf9b920, 0x1ce1e90)
	/usr/local/go/src/runtime/panic.go:969 +0x166
github.com/yuin/gopher-lua.callGFunction(0xc0086ec0b0, 0x0, 0xc0087741c0)
	/Users/caffix/go/pkg/mod/github.com/yuin/gopher-lua@v0.0.0-20200603152657-dc2b0ca8b37e/vm.go:262 +0x1fd
github.com/yuin/gopher-lua.init.3.func26(0xc0086ec0b0, 0xc07c080403, 0xc008742000, 0x0)
	/Users/caffix/go/pkg/mod/github.com/yuin/gopher-lua@v0.0.0-20200603152657-dc2b0ca8b37e/vm.go:817 +0x3b1
github.com/yuin/gopher-lua.mainLoop(0xc0086ec0b0, 0xc008742000)
	/Users/caffix/go/pkg/mod/github.com/yuin/gopher-lua@v0.0.0-20200603152657-dc2b0ca8b37e/vm.go:31 +0xf0
github.com/yuin/gopher-lua.(*LState).callR(0xc0086ec0b0, 0x4, 0x0, 0x0)
	/Users/caffix/go/pkg/mod/github.com/yuin/gopher-lua@v0.0.0-20200603152657-dc2b0ca8b37e/state.go:1202 +0x248
github.com/yuin/gopher-lua.(*LState).Call(...)
	/Users/caffix/go/pkg/mod/github.com/yuin/gopher-lua@v0.0.0-20200603152657-dc2b0ca8b37e/state.go:1954
github.com/yuin/gopher-lua.(*LState).PCall(0xc0086ec0b0, 0x4, 0x0, 0x0, 0x13f6ca0, 0xc033cec400)
	/Users/caffix/go/pkg/mod/github.com/yuin/gopher-lua@v0.0.0-20200603152657-dc2b0ca8b37e/state.go:2017 +0x114
github.com/yuin/gopher-lua.(*LState).CallByParam(0xc0086ec0b0, 0x14110a0, 0xc0087755c0, 0x0, 0x1, 0x0, 0xc022185be8, 0x4, 0x4, 0x40925b, ...)
	/Users/caffix/go/pkg/mod/github.com/yuin/gopher-lua@v0.0.0-20200603152657-dc2b0ca8b37e/state.go:2035 +0xe4
github.com/OWASP/Amass/v3/datasrcs.(*Script).OnSubdomainDiscovered(0xc008720a00, 0x140d8e0, 0xc0080b0000, 0xc03418eba0, 0x1)
	/Users/caffix/go/src/github.com/OWASP/Amass/datasrcs/script.go:327 +0x392
reflect.Value.call(0xf73880, 0xc0156f84a0, 0x13, 0x10f5e52, 0x4, 0xc04b10dec0, 0x3, 0x4, 0xc022185f28, 0x40658a, ...)
	/usr/local/go/src/reflect/value.go:460 +0x8ab
reflect.Value.Call(0xf73880, 0xc0156f84a0, 0x13, 0xc04b10dec0, 0x3, 0x4, 0x0, 0x0, 0x0)
	/usr/local/go/src/reflect/value.go:321 +0xb4
github.com/OWASP/Amass/v3/requests.(*BaseService).processRequests(0xc008720a00)
	/Users/caffix/go/src/github.com/OWASP/Amass/requests/service.go:332 +0x1c6
created by github.com/OWASP/Amass/v3/requests.(*BaseService).Start
	/Users/caffix/go/src/github.com/OWASP/Amass/requests/service.go:130 +0xf8
```

**Comment by fersingb** [comment]
Hello,

I also got something that looks similar with 3.10.5

```
$ amass enum -d domain.com -dir outdir

[....]

panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x30 pc=0xdb40f6]

goroutine 224475 [running]:
github.com/OWASP/Amass/v3/resolvers.(*BaseResolver).tcpExchange(0xc00016f7a0, 0x2aff577, 0xc026648640)
        /home/caffix/go/src/github.com/OWASP/Amass/resolvers/resolver.go:517 +0x56
created by github.com/OWASP/Amass/v3/resolvers.(*BaseResolver).processMessage
        /home/caffix/go/src/github.com/OWASP/Amass/resolvers/resolver.go:482 +0x519
```

**Comment by phackt** [comment]
Same here, i m building like this:
```docker build -t amass https://github.com/OWASP/Amass.git```

 OS info:
```
DISTRIB_ID=Ubuntu
DISTRIB_RELEASE=18.04
DISTRIB_CODENAME=bionic
DISTRIB_DESCRIPTION="Ubuntu 18.04.5 LTS"
```

And got the same error while **bruteforcing** thanks to dic:
```
panic: runtime error: invalid memory address or nil pointer dereference
```

**Comment by phackt** [comment]
```-json /output/phackt.com.json``` is not working, ```-json /output/out.json``` is... weirdo

**Comment by caffix** [comment]
> Hello,
> 
> I also got something that looks similar with 3.10.5
> 
> ```
> $ amass enum -d domain.com -dir outdir
> 
> [....]
> 
> panic: runtime error: invalid memory address or nil pointer dereference
> [signal SIGSEGV: segmentation violation code=0x1 addr=0x30 pc=0xdb40f6]
> 
> goroutine 224475 [running]:
> github.com/OWASP/Amass/v3/resolvers.(*BaseResolver).tcpExchange(0xc00016f7a0, 0x2aff577, 0xc026648640)
>         /home/caffix/go/src/github.com/OWASP/Amass/resolvers/resolver.go:517 +0x56
> created by github.com/OWASP/Amass/v3/resolvers.(*BaseResolver).processMessage
>         /home/caffix/go/src/github.com/OWASP/Amass/resolvers/resolver.go:482 +0x519
> ```

This problem has now been addressed in issue #519
